packages/xrpl/src/models/transactions/AMMBid.ts

```
@@ -0,0 +1,145 @@
+
+  /**
+   * A list of up to 4 additional accounts that you allow to trade at the discounted fee.
+   * This cannot include the address of the transaction sender.
+   */
+  AuthAccounts?: AuthAccount[]
+}
+
+/**
+ * Verify the form and type of an AMMBid at runtime.
+ *
+ * @param tx - An AMMBid Transaction.
+ * @throws When the AMMBid is Malformed.
+ */
+// eslint-disable-next-line max-lines-per-function -- necessary for validateAMMBid
+export function validateAMMBid(tx: Record<string, unknown>): void {
+  validateBaseTransaction(tx)
+
+  if (tx.Asset == null) {
+    throw new ValidationError('AMMBid: missing field Asset')
+  }
+
+  if (!isCurrency(tx.Asset)) {
+    throw new ValidationError('AMMBid: Asset must be an Issue')
+  }
+
+  if (tx.Asset2 == null) {
+    throw new ValidationError('AMMBid: missing field Asset2')
+  }
+
+  if (!isCurrency(tx.Asset2)) {
+    throw new ValidationError('AMMBid: Asset2 must be an Issue')
+  }
+
+  if (tx.BidMin != null && !isAmount(tx.BidMin)) {
+    throw new ValidationError('AMMBid: BidMin must be an Amount')
+  }
+
+  if (tx.BidMax != null && !isAmount(tx.BidMax)) {
+    throw new ValidationError('AMMBid: BidMax must be an Amount')
+  }
+
+  if (tx.AuthAccounts != null) {
+    if (!Array.isArray(tx.AuthAccounts)) {
+      throw new ValidationError(
+        `AMMBid: AuthAccounts must be an AuthAccount array`,
+      )
+    }
+    if (tx.AuthAccounts.length > MAX_AUTH_ACCOUNTS) {
+      throw new ValidationError(
+        `AMMBid: AuthAccounts length must not be greater than ${MAX_AUTH_ACCOUNTS}`,
+      )
+    }
+    if (
+      !isAuthAccounts(
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Only used by JS
+        tx.Account as string,
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Only used by JS
+        tx.AuthAccounts as Array<Record<string, unknown>>,
+      )
+    ) {
+      throw new ValidationError(`AMMBid: invalid AuthAccounts`)
```

FYI you could make this a bit easier to debug by throwing the error in `isAuthAccounts` instead of returning a bool there